js/modules/k6/browser/common/page.go

```diff
@@ -685,11 +689,14 @@ func (p *Page) attachFrameSession(fid cdp.FrameID, fs *FrameSession) {
 	fs.page.frameSessions[fid] = fs
 }
 
-func (p *Page) getFrameSession(frameID cdp.FrameID) *FrameSession {
+func (p *Page) getFrameSession(frameID cdp.FrameID) (*FrameSession, bool) {
 	p.logger.Debugf("Page:getFrameSession", "sid:%v fid:%v", p.sessionID(), frameID)
 	p.frameSessionsMu.RLock()
 	defer p.frameSessionsMu.RUnlock()
-	return p.frameSessions[frameID]
+
+	v, ok := p.frameSessions[frameID]
```

Comment:
Should we also add some logic that will check that `*FrameSession` isn't nil before writing into the `p.frameSessions` map, to be safe against cases like:

```go
p.attachFrameSession(fid, nil)

fs, ok := p. getFrameSession(fid)

// fs = nil
// ok = true
```

What do you think?

Reply:
I don't see anything wrong with being defensive. In our case `attachFrameSession` is called from [here](https://github.com/grafana/k6/blob/88f765b9dafa2749192f415d983960a66708bc47/js/modules/k6/browser/common/frame_session.go#L1027), where the `frameSession` is created on the line above.

Added in 45e8b6a0fd3f13565e6551bed7a0361e9ade6987